### PR TITLE
Fix link state on emulated NICs

### PR DESCRIPTION
**Pull request by jean-edouard**
OXT-886

Signed-off-by: Jed <lejosnej@ainfosec.com>

### CODE REVIEW

File: recipes-openxt/qemu-dm/qemu-dm-2.6.2/nic-link-state-propagation.patch [hunk 1]
```diff
@@ -178,32 +187,31 @@ Index: qemu-2.6.2/hw/xen/xen_backend.c
...
 +    char base[XEN_BUFSIZE];
 +    int val;
 +
 +    assert(opaque);
-+    assert(ni->used);
-+    assert(ni->netdev);
-+    assert(ni->netdev->peer);
++    /* assert(ni->used); */
++    /* assert(ni->netdev); */
++    /* assert(ni->netdev->peer); */
```
Comment (eric-ch) [inline recipes-openxt/qemu-dm/qemu-dm-2.6.2/nic-link-state-propagation.patch]: Dead code.

Comment (eric-ch) [inline recipes-openxt/qemu-dm/qemu-dm-2.6.2/nic-link-state-propagation.patch]: Dead code.

Comment (jean-edouard) [inline recipes-openxt/qemu-dm/qemu-dm-2.6.2/nic-link-state-propagation.patch]: Oh thanks, missed that, will fix

**Comment by jean-edouard** [comment]
Gone

**Comment by jean-edouard** [comment]
Built here: http://openxt-builder.ainfosec.com:8010/builders/openxt/builds/972

**Comment by eric-ch** [comment]
LGTM, +1. Will merge soon.

Tested on a Dell E7470, with a Win10 64b HVM guest, no tools installed:
- Emulated card with its backend in brwireless reports ```disconnected``` states.
- qemu cmdline: ```[...]-device e1000,id=vif0,netdev=net0,mac=ba:d5:22:a9:4a:ff -netdev type=tap,id=net0,ifname=vif7.0-emu,script=/etc/qemu/qemu-ifup,downscript=/etc/qemu/qemu-ifup -device e1000,id=vwif1,netdev=net1,mac=5e:55:c9:87:9f:d2 -netdev type=tap,id=net1,ifname=vif7.1-emu,script=/etc/qemu/qemu-ifup,downscript=/etc/qemu/qemu-ifup```
- xenstore: ```/local/domain/8/device/vif/1/disconnect = "1"``` and ```/local/domain/7/device/vwif/1/disconnect = "1"```

**Comment by eric-ch** [comment]
@jean-edouard @dpsmith OXT-883 is marked for 7.x., if that is correct it would require a stable-7 PR.

**Comment by jean-edouard** [comment]
As mentioned on the ticket, the disconnect nodes don't always get updated properly by xenmgr/nwd.
Without this PR, the disconnect nodes are ignored, so the behavior is deterministic: both interfaces will always be considered connected, even when they're not.
It may or may not be preferable for stable-7 to stay in that state, at least until a xenmgr/nwd fix is found...

**Comment by dpsmith** [comment]
Yes, this should be going into stable-7. Please submit a stable-7 PR. Thanks!

**Comment by jean-edouard** [comment]
Here: https://github.com/OpenXT/xenclient-oe/pull/715

**Pull request merged**